job_creator.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+class JobCreator
+{
+    public function getInstallerVersion($githubRepository, $branch)
+    {
+        $repo = explode('/', $githubRepository)[1];
+        if (in_array($repo, NO_INSTALLER_REPOS)) {
+            return '';
+        }
+        $v = explode('.', $branch);
+        if (count($v) == 1) {
+            return '4.x-dev';
+        }
+        if (in_array($repo, LOCKSTEPED_REPOS)) {
+            return '4' . $v[1] . 'x-dev';
+        } else {
+            // use the latest minor version of installer
+            $a = array_keys(INSTALLER_TO_PHP_VERSIONS);
+            $a = array_map(fn($k) => (int) $k, $a);
+            sort($a);
+            return $a[count($a) - 1] . 'x-dev';
+        }
+    }
+    
+    public function createJob($phpNum, $opts)
```

Comment:
```suggestion
    public function createJob(int $phpIndex, array $opts): array
```
It took me a while to realise that `phpNum` wasn't actually the _version_ of php to use, but it's the _index in the array_ to choose from. This new variable name makes that clear up front.

Reply:
Updated